client/windows/CMarketWindow.cpp

```diff
@@ -192,13 +200,15 @@ void CMarketWindow::createArtifactsSelling(const IMarket * market, const CGHeroI
 	// Create image that copies part of background containing slot MISC_1 into position of slot MISC_5
 	artSlotBack = std::make_shared<CPicture>(background->getSurface(), Rect(20, 187, 47, 47), 0, 0);
 	artSlotBack->moveTo(pos.topLeft() + Point(18, 339));
-	auto artsSellingMarket = std::make_shared<CArtifactsSelling>(market, hero);
+	const auto mapObj = LOCPLINT->cb->getObj(market->getObjInstanceID());
+	auto artsSellingMarket = std::make_shared<CArtifactsSelling>(market, hero, LOCPLINT->cb->getTown(mapObj->id) ?
+		VLC->generaltexth->translate("building.core.conflux.special1.name") : mapObj->getObjectName());
```

Comment:
Perhaps we should instead provide this string from outside somehow? Or provide its building ID?

Reply:
Added `CMarketWindow::getMarketTitle` func